internal/tools/aws/ec2.go

```diff
@@ -163,3 +167,35 @@ func GetSecurityGroupsWithFilters(filters []*ec2.Filter) ([]*ec2.SecurityGroup,
 
 	return sgOutput.SecurityGroups, nil
 }
+
+// IsValidAMI check if the provided AMI exists
+func (a *Client) IsValidAMI(AMIImage string) bool {
```

Comment:
I think it may make more sense to return `(bool, error)` here instead of logging within the function. No logger is passed in so we wouldn't have all the extra logging fields that get created before this is called. If we return errors we can log them a bit better.